src/test/java/com/commercetools/sync/categories/helpers/CategoryReferenceResolverTest.java

```diff
@@ -51,19 +58,70 @@ public void setup() {
         categoryService = mock(CategoryService.class);
         when(categoryService.fetchCachedCategoryId(CACHED_CATEGORY_KEY))
             .thenReturn(CompletableFuture.completedFuture(Optional.of(CACHED_CATEGORY_ID)));
-        syncOptions = CategorySyncOptionsBuilder.of(mock(SphereClient.class)).build();
+        final CategorySyncOptions syncOptions = CategorySyncOptionsBuilder.of(mock(SphereClient.class)).build();
+        referenceResolver = new CategoryReferenceResolver(syncOptions, typeService, categoryService);
+    }
+
+    @Test
+    public void resolveAssetsReferences_WithNullAssets_ShouldNotResolveAssets() {
+        final CategoryDraftBuilder categoryDraftBuilder =
+            getMockCategoryDraftBuilder(Locale.ENGLISH, "myDraft", "key", CACHED_CATEGORY_KEY,
+                "customTypeId", new HashMap<>())
+                .assets(null);
+
+        final CategoryDraftBuilder resolvedBuilder = referenceResolver.resolveAssetsReferences(categoryDraftBuilder)
+                                                                      .toCompletableFuture().join();
+
+        final List<AssetDraft> resolvedBuilderAssets = resolvedBuilder.getAssets();
+        assertThat(resolvedBuilderAssets).isNull();
```

Comment:
Maybe the function "getAsset" should return a empty collect instead of null

Reply:
if the assets are not given to the JVM SDK AssetDraftBuilder. The assets would be `null`. Thanks for reminding me tho to add a unit test for empty case: eb4e50a